next-docs/pages/guides/hello-world/add-system.mdx

```diff
@@ -0,0 +1,296 @@
+
+## Setup
+
+[Create a new MUD application from the template](../../templates/typescript/getting-started).
+
+## Add a contract for the new system
+
+Create a file `packages/contracts/src/systems/DecrementSystem.sol`.
+Note that we could have just added a function to the existing system, `IncrementSystem.sol`.
+The only reason we are adding a new system here is to see how to do it.
+
+```solidity filename="DecrementSystem.sol" copy showLineNumbers
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import { System } from "@latticexyz/world/src/System.sol";
+import { Counter } from "../codegen/index.sol";
+
+contract DecrementSystem is System {
+  function decrement() public returns (uint32) {
+    uint32 counter = Counter.get();
+    uint32 newValue = counter - 1;
+    Counter.set(newValue);
+    return newValue;
+  }
+}
+```
+
+<details>
+<summary>Explanation</summary>
+
+<p></p>
+
+```solidity
+import { System } from "@latticexyz/world/src/System.sol";
+import { Counter } from "../codegen/index.sol";
+```
+
+The two things the system needs to know: how to be a `System` and how to access the `Counter`.
+
+```solidity
+uint32 counter = Counter.get();
+```
+
+Get the counter value.
+
+```solidity
+Counter.set(newValue);
+```
+
+Set the counter to a new value.
+
+</details>
+
+## Add `decrement` to the application
+
+Having a system be able to do something doesn't help anybody unless it is called from somewhere.
+In this case, the vanilla getting started front end.
+
+1.  Edit `packages/client/src/mud/createSystemCalls.ts` to include `decrement`.
+    This is the file after the changes:
+
+    <CollapseCode>
+
+        ```typescript filename="createSystemCalls.ts" copy showLineNumbers {47-51,55}
+        /*
+        * Create the system calls that the client can use to ask
+        * for changes in the World state (using the System contracts).
+        */
+
+        import { getComponentValue } from "@latticexyz/recs";
+        import { ClientComponents } from "./createClientComponents";
+        import { SetupNetworkResult } from "./setupNetwork";
+        import { singletonEntity } from "@latticexyz/store-sync/recs";
+
+        export type SystemCalls = ReturnType<typeof createSystemCalls>;
+
+        export function createSystemCalls(
+          /*
+          * The parameter list informs TypeScript that:
+          *
+          * - The first parameter is expected to be a
+          *   SetupNetworkResult, as defined in setupNetwork.ts
+          *
+          * - Out of this parameter, we only care about two fields:
+          *   - worldContract (which comes from createContract, see
+          *     https://github.com/latticexyz/mud/blob/26dabb34321eedff7a43f3fcb46da4f3f5ba3708/templates/vanilla/packages/client/src/mu$
+          *   - waitForTransaction (which comes from syncToRecs, see
+          *     https://github.com/latticexyz/mud/blob/26dabb34321eedff7a43f3fcb46da4f3f5ba3708/templates/vanilla/packages/client/src/mu$
```

Comment:
these two links seem to be cut off. Also, should we link to the most recent version on `main` instead of the permalink to a previous state of the repo?

Reply:
Oops on the links. The problem with linking to the latest version is that line numbers could change (and those are part of the real link). But then again, the template doesn't change that often - I'll use main and people can just search if the information isn't at the exact line.